### PR TITLE
add kube-storage-version-migrator

### PR DESCRIPTION
**Pull request by umezawatakeshi**
part of https://github.com/cybozu-go/neco/issues/1909

Signed-off-by: UMEZAWA Takeshi <takeshi-umezawa@cybozu.co.jp>

### CODE REVIEW

**Comment by ysksuzuki** [comment]
Did you check if k8s-storage-version-migration is working as expected manually? Or are we going to see it in our staging environment?

**Comment by umezawatakeshi** [comment]
I tested on dctest environment. cf. https://bozuman.cybozu.com/k/#/space/3801/thread/54732/3272766

File: argocd-config/base/kube-storage-version-migrator.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  - resources-finalizer.argocd.argoproj.io
+spec:
+  project: default
+  source:
+    repoURL: https://github.com/cybozu-go/neco-apps.git
+    targetRevision: release
+    path: kube-storage-version-migrator/base
+  destination:
+    server: https://kubernetes.default.svc
+    namespace: kube-storage-version-migrator
```
Comment (zoetrope) [inline argocd-config/base/kube-storage-version-migrator.yaml:19]: I think this namespace is a bit long, but it's not a big problem.

File: docs/maintenance.md [hunk 2]
```diff
@@ -141,6 +142,15 @@ $ make update-kube-metrics-adapter
 $ git diff kube-metrics-adapter
 ```
 
+## kube-storage-version-migrator
+
+1. Check the [release page](https://github.com/kubernetes-sigs/kube-storage-version-migrator/releases) for changes.
```
Comment (zoetrope) [inline docs/maintenance.md:147]: We can use pre-release? (A note might be helpful.)

File: monitoring/base/victoriametrics/rules/kube-storage-version-migrator-scrape.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+apiVersion: operator.victoriametrics.com/v1beta1
+kind: VMPodScrape
+metadata:
+  name: kube-storage-version-migrator-migrator
```
Comment (zoetrope) [inline monitoring/base/victoriametrics/rules/kube-storage-version-migrator-scrape.yaml:4]: Why do you repeat `migrator` twice?

Comment (umezawatakeshi) [reply monitoring/base/victoriametrics/rules/kube-storage-version-migrator-scrape.yaml:4]: The last part stands for Pod name. If `trigger` Pod expose its metrics in the future, the name of `VMPodScrape` will be `kube-storage-version-migrator-trigger`.

File: docs/maintenance.md (hunk 2 above)
Comment (umezawatakeshi) [reply docs/maintenance.md:147]: sure.

**Review by zoetrope — APPROVED** [review]

**Review by yokaze — APPROVED** [review]

**Review by ysksuzuki — APPROVED** [review]

**Pull request merged**